src/Hosting/TestHost/src/HttpContextBuilder.cs

```diff
@@ -59,15 +60,18 @@ internal void Configure(Action<HttpContext> configureContext)
         /// Start processing the request.
         /// </summary>
         /// <returns></returns>
-        internal Task<HttpContext> SendAsync(CancellationToken cancellationToken)
+        internal Task<HttpContext> SendAsync(CancellationToken cancellationToken, bool preserveExecutionContext)
```

Comment:
preserveExecutionContext should be a property like AllowSynchronousIO.

Reply:
I disagree. It's only used in this method and all the callers have a value they can provide here. `AllowSynchronousIO` is only a property because `HttpContextBuilder` implements `IHttpBodyControlFeature` and requires that the property be provided. In fact, no caller uses the setter on `HttpContextBuilder`, they only use the constructor argument.

Reply:
And why not make it a constructor argument?

Reply:
🤷‍♀️